### PR TITLE
Issue-794

### PR DESCRIPTION
**Pull request by ramapalani**
Enable Zalenium UI pages to be served from a non root context

Also change tab in secret.yaml to two spaces

### CODE REVIEW

File: charts/zalenium/templates/ingress.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spec:
   - host: {{ .Values.ingress.hostname }}
     http:
       paths:
-      - path: /
+      - path: /{{ .Values.ingress.contextpath }}
```
Comment (arnaud-deprez) [inline charts/zalenium/templates/ingress.yaml]: I would prefer to have the / part of the injected value if possible and rename the variable to path so it's a one on one match with the actual ingress configuration:

```suggestion
      - path: {{ .Values.ingress.path }}
```

File: charts/zalenium/values.yaml [hunk 2]
```diff
@@ -143,15 +143,20 @@ persistence:
...
 
 ingress:
-  enabled: false
+  enabled: true
   tls: false
   # secretName: my-tls-cert # only needed if tls above is true
   hostname: zalenium.foobar.com
+  # context path without the forward slash (/)
+  # leaving this empty would run zalenium in the root context (/)
+  contextpath:
```
Comment (arnaud-deprez) [inline charts/zalenium/values.yaml]: Then put a default value so it is just easier to enable/disable the ingress: 
```suggestion
  contextpath: /
```

File: charts/zalenium/values.yaml [hunk 3]
```diff
@@ -143,15 +143,20 @@ persistence:
     size: 80Gi
 
 ingress:
-  enabled: false
+  enabled: true
```
Comment (arnaud-deprez) [inline charts/zalenium/values.yaml]: Most of the public chart tend to disable it by default as in some cloud provider, it is just easier to play with a service of type `LoadBalancer`. I think it is better to follow this kind of guideline.

**Comment by ramapalani** [comment]
*@arnaud-deprez *thanks for reviewing the code. I'll make these changes and
update the PR.

**Comment by diemol** [comment]
Thanks for the review @arnaud-deprez 👍

File: charts/zalenium/templates/ingress.yaml (hunk 1 above)
Comment (ramapalani) [reply charts/zalenium/templates/ingress.yaml]: Made this change

File: charts/zalenium/values.yaml (hunk 2 above)
Comment (ramapalani) [reply charts/zalenium/values.yaml]: Made this change

**Comment by ramapalani** [comment]
@arnaud-deprez  Can you review again?

File: charts/zalenium/values.yaml [hunk 4]
```diff
@@ -147,11 +147,15 @@ ingress:
   tls: false
   # secretName: my-tls-cert # only needed if tls above is true
   hostname: zalenium.foobar.com
+  # context path without the forward slash (/)
+  # leaving this empty would run zalenium in the root context (/)
+  contextpath: /
   annotations:
-    # kubernetes.io/ingress.class: "nginx"
+    kubernetes.io/ingress.class: "nginx"
```
Comment (arnaud-deprez) [inline charts/zalenium/values.yaml]: please, leave this line commented as no everyone is using nginx ingress.
```suggestion
    # kubernetes.io/ingress.class: "nginx"
```

File: charts/zalenium/templates/_pod-template.yaml [hunk 5]
```diff
@@ -107,6 +107,8 @@ spec:
           value: {{ .Values.hub.keepOnlyFailedTests | quote }}
         - name: RETENTION_PERIOD
           value: {{ .Values.hub.retentionPeriod | quote }}
+        - name: CONTEXT_PATH
+          value: {{ .Values.ingress.contextpath | quote }}
```
Comment (arnaud-deprez) [inline charts/zalenium/templates/_pod-template.yaml]: Can you please use the same name `path` as the Kubernetes Ingress object. It is just more intuitive IMO.
```suggestion
           value: {{ .Values.ingress.path | quote }}
```

File: charts/zalenium/templates/ingress.yaml [hunk 6]
```diff
@@ -18,7 +18,7 @@ spec:
   - host: {{ .Values.ingress.hostname }}
     http:
       paths:
-      - path: /
+      - path: {{ .Values.ingress.contextpath }}
```
Comment (arnaud-deprez) [inline charts/zalenium/templates/ingress.yaml]: Same remark: use the same name `path` as the Kubernetes Ingress object.

File: charts/zalenium/values.yaml [hunk 7]
```diff
@@ -147,11 +147,15 @@ ingress:
   tls: false
   # secretName: my-tls-cert # only needed if tls above is true
   hostname: zalenium.foobar.com
+  # context path without the forward slash (/)
+  # leaving this empty would run zalenium in the root context (/)
+  contextpath: /
```
Comment (arnaud-deprez) [inline charts/zalenium/values.yaml]: Same remark: Can you please use the same name `path` as the Kubernetes Ingress object.

**Comment by diemol** [comment]
Thanks @ramapalani for your contribution! 

Thanks @arnaud-deprez for the review 👍

**Pull request merged**